.github/workflows/roman_ci.yml

```diff
@@ -14,110 +14,85 @@ on:
+      CRDS_PATH: /tmp/crds_cache
     steps:
-      - if: contains(runner.os, 'Linux') && github.event_name != 'pull_request'
-        uses: datadog/agent-github-action@v1
-        with:
-          api_key: ${{ secrets.DD_API_KEY }}
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - uses: actions/setup-python@v4
-        with:
-          python-version: ${{ matrix.python-version }}
-          cache: 'pip'
-          cache-dependency-path: 'pyproject.toml'
-      - run: pip install "tox<4.0"
-      - run: >
+      - id: context
+        run: >
           echo "pmap=$(
-          curl -s -X POST -d '{"jsonrpc": "1.0", "method": "get_default_context", "params": ["roman"], "id": 1}' ${{ env.CRDS_SERVER_URL }}/json/ |
+          curl -s -X POST -d '{"jsonrpc": "1.0", "method": "get_default_context", "params": ["${{ env.OBSERVATORY }}"], "id": 1}' ${{ env.CRDS_SERVER_URL }}/json/ |
           python -c "import sys, json; print(json.load(sys.stdin)['result'])"
           )" >> $GITHUB_OUTPUT
         # Get default CRDS_CONTEXT without installing crds client
         # See https://hst-crds.stsci.edu/static/users_guide/web_services.html#generic-request
-        id: crds-context
-      - uses: actions/cache@v3
-        with:
-          path: ${{ env.CRDS_PATH }}
-          key: crds-${{ steps.crds-context.outputs.pmap }}
-      - run: tox -e ${{ matrix.toxenv }}-ddtrace
-        env:
-          DD_GIT_REPOSITORY_URL: ${{ github.repositoryUrl }}
-          DD_GIT_COMMIT_SHA: ${{ github.sha }}
-          DD_GIT_BRANCH: ${{ github.ref_name }}
-      - if: contains(matrix.toxenv, '-cov')
-        uses: codecov/codecov-action@v3
-        with:
-          file: ./coverage.xml
-          flags: unit
-          fail_ci_if_error: true
+      - id: path
+        run: echo "path=${{ env.CRDS_PATH }}" >> $GITHUB_OUTPUT
+      - id: server
+        run: echo "url=${{ env.CRDS_SERVER_URL }}" >> $GITHUB_OUTPUT
+    outputs:
+      context: ${{ steps.context.outputs.pmap }}
+      path: ${{ steps.path.outputs.path }}
+      server: ${{ steps.server.outputs.url }}
+  check:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@v1
+    with:
+      envs: |
+        - linux: check-style
+        - linux: check-dependencies
+        - linux: build-dist
+  test:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    needs: [ crds ]
+    with:
+      setenv: |
+        CRDS_PATH: ${{ needs.crds.outputs.path }}
+        CRDS_SERVER_URL: ${{ needs.crds.outputs.server }}
+        CRDS_CLIENT_RETRY_COUNT: 3
+        CRDS_CLIENT_RETRY_DELAY_SECONDS: 20
+        DD_SERVICE: romancal
+        DD_ENV: ci
+        DD_GIT_REPOSITORY_URL: ${{ github.repositoryUrl }}
+        DD_GIT_COMMIT_SHA: ${{ github.sha }}
+        DD_GIT_BRANCH: ${{ github.ref_name }}
+      cache-path: /tmp/crds_cache
+      cache-key: crds-${{ needs.crds_context.outputs.pmap }}
```

Comment:
The lack of a public CRDS server means that when run on github this creates a key `crds-`.
https://github.com/spacetelescope/romancal/actions/runs/4440790860/jobs/7795041648?pr=612#step:3:9
I'm not sure this is the cause of the `Warning: Path Validation Error` (or what that means):
https://github.com/spacetelescope/romancal/actions/runs/4440790860/jobs/7795041648?pr=612#step:21:2
I also see no cache with the key `crds-`. Are any of these issues?

Reply:
this fails now because of the lack of a server, but it should start to work when the server is available and CRDS is enabled in the CI tests for `romancal`